### PR TITLE
Fix misnamed parameters for model architectures

### LINKED ISSUE #19: Correct ChemPropFeaturizer param name from `ffn_hidden_num_layers` to `ffn_hidden_layers`
`ffn_hidden_num_layers` is incorrectly named and silently dropped by Pydantic.

### PR DESCRIPTION
**Pull request by cynthiaxu04**
Resolves #19 

Also checks and fixes any other misnamed parameters for other model architectures besides ChemProp & CheMeleon. Two issues found:

1.   LGBM (single_task/lgbm and ensemble/single_task/lgbm): alpha → reg_alpha
LGBMModelBase defines reg_alpha as the L1 regularization field. Since it has no extra="allow", alpha was silently dropped and never passed to LightGBM.

2. TabPFN (single_task/tabpfn and ensemble/single_task/tabpfn): device: cuda → accelerator: cuda
  TabPFNModelBase exposes accelerator (accepting "cpu", "cuda", "auto"), not device. The device key was silently ignored, leaving it defaulting to auto.

### CODE REVIEW

File: canonical_recipes/ensemble/single_task/lgbm/lgbm.yaml [hunk 1]
```diff
@@ -49,7 +51,7 @@ procedure:
     type: LGBMRegressorModel
     # Specify model parameters
     params:
-      alpha: 0.005
+      reg_alpha: 0.005
```
Comment (smcolby) [inline canonical_recipes/ensemble/single_task/lgbm/lgbm.yaml:56]: What's the difference here? Not familiar with LGBMs so much

**Review by smcolby — CHANGES_REQUESTED** [review]
* For ensemble models, use_bagging should be True. This was not changed.
* Ensembles should also have the bagging seed specified (so users know it exists from the template). It's `bootstrap_seed` in the `ensemble` section of the YAML.
* It doesn't look like you changed the TabPFN CV config to match other configs
* "Tree-based recipes use 0.8/0.0/0.2; DL recipes use 0.7/0.1/0.2. Under the same seed the test molecules differ, making cross-model metric comparisons invalid." -- can we confirm that the "0.2" is the same in both of these split configuration?

**Comment by cynthiaxu04** [comment]
>     * It doesn't look like you changed the TabPFN CV config to match other configs

the ensemble single task TabPFN has no CV config because it's ensemble. but the regular single task TabPFN does have updated CV config to 5x5 now

**Comment by cynthiaxu04** [comment]
> * "Tree-based recipes use 0.8/0.0/0.2; DL recipes use 0.7/0.1/0.2. Under the same seed the test molecules differ, making cross-model metric comparisons invalid." -- can we confirm that the "0.2" is the same in both of these split configuration?

Confirmed, the 0.2 test set is the same in both split cases. ANVIL splits the test set first, before train and/or val

File: canonical_recipes/ensemble/single_task/lgbm/lgbm.yaml (hunk 1 above)
Comment (cynthiaxu04) [reply canonical_recipes/ensemble/single_task/lgbm/lgbm.yaml:56]: `alpha` is the dead parameter name for LGBM. `reg_alpha` is the correct parameter name for L1 regularization for LGBM. `alpha` is the correct parameter name for L1 regularization for XGBoost. Comments have been added to clarify this

**Review by smcolby — APPROVED** [review]
LGTM
